app/controllers/detail-viewers.ts

```diff
@@ -89,10 +90,39 @@ function setupAnswersReasonsForAppeal(req: Request): Array<any> {
     const listOfDocuments: string[] = data.reasonsForAppealDocuments.map(evidence => {
       const fileId = docStoreUrlToId(evidence.value.document_url, req.session.appeal.documentMap);
       const formattedFileName = fileNameFormatter(evidence.value.document_filename);
-      return `<a class='govuk-link' target='_blank' rel='noopener noreferrer' href='${paths.common.documentViewer}/${fileId}'>${formattedFileName}</a>`;
+      return `<a class='govuk-link' target='_blank' rel='noopener noreferrer' href='${paths.common.detailsViewers.document}/${fileId}'>${formattedFileName}</a>`;
     });
-    array.push(addSummaryRow(i18n.pages.overviewPage.timeline.reasonsForAppealCheckAnswersHistory.whyYouThinkHomeOfficeIsWrong, [ data.reasonsForAppealDecision ], null));
-    array.push(addSummaryRow(i18n.pages.overviewPage.timeline.reasonsForAppealCheckAnswersHistory.providingEvidence, [ ...Object.values(listOfDocuments) ], null, Delimiter.BREAK_LINE));
+    array.push(addSummaryRow(i18n.pages.detailViewers.reasonsForAppealCheckAnswersHistory.whyYouThinkHomeOfficeIsWrong, [ data.reasonsForAppealDecision ], null));
+    array.push(addSummaryRow(i18n.pages.reasonsForAppealUpload.title, [ ...Object.values(listOfDocuments) ], null, Delimiter.BREAK_LINE));
+  }
+  return array;
+}
+
+function setupTimeExtensionDecision(req: Request, timeExtensionEvent: TimeExtensionCollection) {
+  const array = [];
+  const data = timeExtensionEvent.value;
+  if (_.has(data, 'decision')) {
+    array.push(addSummaryRow(i18n.pages.detailViewers.timeExtensionReview.decision, [ data.decision ], null));
+  }
+  if (_.has(data, 'decisionReason')) {
+    array.push(addSummaryRow(i18n.pages.detailViewers.timeExtensionReview.reason, [ data.decisionReason ], null));
+  }
+  return array;
+}
+
+function setupTimeExtension(req: Request, timeExtensionEvent: TimeExtensionCollection) {
+  const array = [];
+  const data = timeExtensionEvent.value;
+  if (_.has(data, 'reason')) {
+    array.push(addSummaryRow(i18n.pages.detailViewers.timeExtensionRequest.question, [ data.reason ], null));
+  }
+  if (_.has(data, 'evidence')) {
+    const listOfDocuments: string[] = data.evidence.map(evidence => {
+      const fileId = addToDocumentMapper(evidence.value.document_url, req.session.appeal.documentMap);
+      const formattedFileName = fileNameFormatter(evidence.value.document_filename);
+      return `<a class='govuk-link' target='_blank' rel='noopener noreferrer' href='${paths.common.detailsViewers.document}/${fileId}'>${formattedFileName}</a>`;
```

Comment:
Not for now as there is another branch also doing this but we should probably pull out a detailsViewer util to generate this link as its in loads of controllers now.

Reply:
I agree with should move the logic to utils, I believe I created one already somewhere I'd have a look.

Reply:
I agree with this change so I've refactored the setup methods a bit and abstracted the logic elsewhere